### PR TITLE
Internal API endpoint for clearing access cache

### PR DESCRIPTION
**Pull request by jonkjetiloye**
## Description
- Added new endpoint along-side existing internal rights operations
- Added implementation in Altinn2RightService and Altinn2RightsClient
- Added Bruno manual test requests

## Related Issue(s)
- #648

## Developer/Reviewer Checklist
- [x] **Your** code builds clean without any errors or warnings
- [x] No changes to config/appsettings or environment variables created in separate linked PR
- [x] Manual testing done (required)
- [x] Relevant integration test added (required for functional changes)
- [ ] Relevant automated test added (required for functional changes)
- [x] All integration tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: src/Altinn.AccessManagement.Core/Services/Altinn2RightsService.cs [hunk 1]
```diff
@@ -35,6 +44,45 @@ public async Task<List<RightDelegation>> GetReceivedRights(int partyId, Cancella
         return await MapDelegationResponse(delegations);
     }
 
+    /// <inheritdoc/>
+    public async Task<HttpResponseMessage> ClearReporteeRights(int fromPartyId, BaseAttribute toAttribute, CancellationToken cancellationToken = default) => toAttribute.Type switch
```
Comment (andreasisnes) [inline src/Altinn.AccessManagement.Core/Services/Altinn2RightsService.cs:48]: funksjonelt er nice da

File: src/Altinn.AccessManagement/Controllers/RightsInternalController.cs [hunk 2]
```diff
@@ -388,5 +387,33 @@ public async Task<IActionResult> RevokeOfferedDelegation([FromRoute, FromHeader]
...
+        [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status500InternalServerError)]
+        [FeatureGate(FeatureFlags.RightsDelegationApi)]
+        public async Task<IActionResult> ClearAccessCache([FromRoute] int party, [FromBody] BaseAttributeExternal to, CancellationToken cancellationToken)
+        {
+            BaseAttribute toAttribute = _mapper.Map<BaseAttribute>(to);
+            HttpResponseMessage response = await _rightsForAltinn2.ClearReporteeRights(party, toAttribute, cancellationToken);
+            if (!response.IsSuccessStatusCode)
+            {
+                return new ObjectResult(ProblemDetailsFactory.CreateProblemDetails(HttpContext, (int)response.StatusCode, detail: $"Could not complete the request. Reason: {response.ReasonPhrase}"));
```
[the remark this answers is not in the document]
Comment (andreasisnes) [reply src/Altinn.AccessManagement/Controllers/RightsInternalController.cs:413]: however, you should be able to specify the statuscode to the ObjectResult

File: test/Altinn.AccessManagement.Tests/Controllers/Altinn2RightsControllerTest.cs [hunk 3]
```diff
@@ -114,10 +116,90 @@ public async Task GetReceviedDelegations_ReturnOk(string header, string value, A
...
     };
 
+    /// <summary>
+    /// Tests <see cref="RightsInternalController.ClearAccessCache(int, BaseAttributeExternal, System.Threading.CancellationToken)"/>
+    /// </summary>
+    [Theory]
+    [MemberData(nameof(ClearAccessCache_ReturnOk_input))]
+    public async Task ClearAccessCache_ReturnOk(string authnUserToken, int party, BaseAttributeExternal toAttribute, Action<HttpResponseMessage> assert)
+    {
+        var client = NewClient(NewServiceCollection(WithServiceMoq), [WithClientRoute("accessmanagement/api/v1/")]);
```
Comment (andreasisnes) [inline test/Altinn.AccessManagement.Tests/Controllers/Altinn2RightsControllerTest.cs:130]: [WithClientRoute("accessmanagement/api/v1/")]

do you need the brackets here?

File: test/Altinn.AccessManagement.Tests/Controllers/Altinn2RightsControllerTest.cs [hunk 4]
```diff
@@ -114,10 +116,90 @@ public async Task GetReceviedDelegations_ReturnOk(string header, string value, A
...
+    };
+
+    /// <summary>
+    /// Tests <see cref="RightsInternalController.ClearAccessCache(int, BaseAttributeExternal, System.Threading.CancellationToken)"/>
+    /// </summary>
+    [Theory]
+    [MemberData(nameof(ClearAccessCache_ReturnBadRequest_input))]
+    public async Task ClearAccessCache_ReturnBadRequest(string authnUserToken, int party, BaseAttributeExternal toAttribute, Action<HttpResponseMessage> assert)
+    {
+        var client = NewClient(NewServiceCollection(WithServiceMoq), [WithClientRoute("accessmanagement/api/v1/")]);
```
Comment (andreasisnes) [inline test/Altinn.AccessManagement.Tests/Controllers/Altinn2RightsControllerTest.cs:173]: brackets?

File: test/Altinn.AccessManagement.Tests/Controllers/Altinn2RightsControllerTest.cs [hunk 5]
```diff
@@ -114,10 +116,90 @@ public async Task GetReceviedDelegations_ReturnOk(string header, string value, A
...
+        {
+            PrincipalUtil.GetToken(20000490, 50002598, 3), // Kasper B�rstad
+            50005545, // From �rsta
+            new BaseAttributeExternal { Type = Urn.Altinn.EnterpriseUser.Uuid, Value = "00000000-0000-0000-0002-000000010727" }, // To OrstaECUser
+            AssertStatusCode(HttpStatusCode.OK)
+        }
+    };
+
+    /// <summary>
+    /// Tests <see cref="RightsInternalController.ClearAccessCache(int, BaseAttributeExternal, System.Threading.CancellationToken)"/>
```
Comment (andreasisnes) [inline test/Altinn.AccessManagement.Tests/Controllers/Altinn2RightsControllerTest.cs:167]: I like this reference style :+1: easy to jump to the executing action

**Review by andreasisnes — APPROVED** [review]

**Review by andreasisnes — APPROVED** [review]

File: src/Altinn.AccessManagement/Controllers/RightsInternalController.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (jonkjetiloye) [reply src/Altinn.AccessManagement/Controllers/RightsInternalController.cs:413]: Magic under the hood stuff, makes the mapping from ProblemDetail to ObjectResult set the correct HttpStatusCode based on the ProblemDetail. So it just works 👍

File: test/Altinn.AccessManagement.Tests/Controllers/Altinn2RightsControllerTest.cs (hunk 3 above)
Comment (jonkjetiloye) [reply test/Altinn.AccessManagement.Tests/Controllers/Altinn2RightsControllerTest.cs]: Nope, you are correct my good sir. I shalt have them removed

File: test/Altinn.AccessManagement.Tests/Controllers/Altinn2RightsControllerTest.cs (hunk 4 above)
Comment (jonkjetiloye) [reply test/Altinn.AccessManagement.Tests/Controllers/Altinn2RightsControllerTest.cs]: not any more

**Pull request merged**